core/finality-grandpa/src/lib.rs

```diff
@@ -872,9 +874,13 @@ impl<B, E, Block: BlockT<Hash=H256>, RA, PRA> BlockImport<Block>
 					self.inner.backend().blockchain(),
 					BlockId::Hash(parent_hash),
 					BlockId::Hash(*base),
-				)?;
+				);
+
+                if tree_route.is_err() {
```

Comment:
I did not mark all space usages. Here are more. Could you please check all of your changes :)

Reply:
Will do!